src/hotspot/share/prims/jvmti.xml

```diff
@@ -12892,15 +12892,21 @@ myInit() {
          id="ThreadStart" const="JVMTI_EVENT_THREAD_START" num="52" phase="start">
     <description>
       A thread start event is generated by a new thread before its initial
-      method executes. The capability
+      method executes.
+      <p/>
+      This event is generated by platform threads. It is also generated by
+      virtual threads when the capability
```

Comment:
Is there a reason you chose to use "by" instead of "for"?  Although the thread that the event is generated "by" is the same as the thread it is generated "for", to me using "by" just reads kind of funny. The consumer of this event is usually not concerned about the thread the event is delivered on, only the thread it is delivered for.

Reply:
> Is there a reason you chose to use "by" instead of "for"? 

The specification of these two events have always used "generated by" in the first sentence of the description so it's just keeping it consistent. The thread lifecycle events are cases where it's important to be guaranteed that the events are reported on the threads themselves.